remove 'when is defined' in main.yml

jsl6ul opened the pull request:
remove conditions 'when is defined' for cobbler_distros, cobbler_profiles and cobbler_systems, to avoid silent failure.

You have to define 'cobbler_systems: []' to maintain the same behavior, but if, for any reason, you have an undefined var in cobbler_distros or cobbler_profiles, the task will not fail silently.

jsl6ul committed:
remove 'when is defined' in main.yml

remove conditions 'when is defined' for cobbler_distros,
cobbler_profiles and cobbler_systems, to avoid silent failure.

You have to define 'cobbler_systems: []' to maintain the same behavior,
but if, for any reason, you have an error in cobbler_distros or
cobbler_profiles, the task will not fail silently.

btravouillon reviewed (changes requested):
This change makes the variables mandatory. Could you add a first task with an assert that ensure the variables are defined? If not, print a message to the user.

jsl6ul commented:
> This change makes the variables mandatory. Could you add a first task with an assert that ensure the variables are defined? If not, print a message to the user.

we should probably define them, empty, in defaults/main. (no breaking change)

jsl6ul committed:
update defaults/main.yml

set this vars empty as default.
(to avoid a breaking change caused by deleting "when is defined")

cobbler_distros: []
cobbler_profiles: []
cobbler_systems: []

jsl6ul committed:
Merge branch 'main' into whendefined

btravouillon reviewed (approved):
Looks good to me. 👍
